core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -238,127 +244,146 @@ trait Endpoint[A] { self =>
+    */
+  final def shouldNot(rule: String)(predicate: A => Boolean)(implicit F: MonadError[F, Throwable]): Endpoint[F, A] =
     should(s"not $rule.")(x => !predicate(x))
 
   /**
-   * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
-   * to be reused across multiple endpoints.
-   *
-   * @param rule the predefined [[ValidationRule]] that will return true if the data is
-   *             valid
-   *
-   * @return an endpoint that will return the value of this reader if it is valid.
-   *         Otherwise the future fails with an [[Error.NotValid]] error.
-   */
-  final def should(rule: ValidationRule[A]): Endpoint[A] = should(rule.description)(rule.apply)
+    * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
+    * to be reused across multiple endpoints.
+    *
+    * @param rule the predefined [[ValidationRule]] that will return true if the data is
+    *             valid
+    *
+    * @return an endpoint that will return the value of this reader if it is valid.
+    *         Otherwise the future fails with an [[Error.NotValid]] error.
+    */
+  final def should(rule: ValidationRule[A])(implicit F: MonadError[F, Throwable]): Endpoint[F, A] =
+    should(rule.description)(rule.apply)
 
   /**
-   * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
-   * to be reused across multiple endpoints.
-   *
-   * @param rule the predefined [[ValidationRule]] that will return false if the data is
-   *             valid
-   *
-   * @return an endpoint that will return the value of this reader if it is valid.
-   *         Otherwise the future fails with a [[Error.NotValid]] error.
-   */
-  final def shouldNot(rule: ValidationRule[A]): Endpoint[A] = shouldNot(rule.description)(rule.apply)
+    * Validates the result of this endpoint using a predefined `rule`. This method allows for rules
+    * to be reused across multiple endpoints.
+    *
+    * @param rule the predefined [[ValidationRule]] that will return false if the data is
+    *             valid
+    *
+    * @return an endpoint that will return the value of this reader if it is valid.
+    *         Otherwise the future fails with a [[Error.NotValid]] error.
+    */
+  final def shouldNot(rule: ValidationRule[A])(implicit F: MonadError[F, Throwable]): Endpoint[F, A] =
+    shouldNot(rule.description)(rule.apply)
 
   /**
     * Lifts this endpoint into one that always succeeds, with [[Try]] representing both success and
     * failure cases.
     */
-  final def liftToTry: Endpoint[Try[A]] =
-    new Endpoint[Try[A]] with (Try[Output[A]] => Output[Try[A]]) {
-      final def apply(toa: Try[Output[A]]): Output[Try[A]] = toa match {
-        case Return(oo) => oo.map(Return.apply)
-        case t @ Throw(_) => Output.payload(t.cast[A])
+  final def liftToTry(implicit F: MonadError[F, Throwable]): Endpoint[F, Try[A]] = attempt.map({
```

Comment:
We should deprecate this and point people to `attempt`.

This may or may not happen as part of this PR - whatever you feel appropriate.

Reply:
Let's deprecate it as a separate PR

Reply:
I agree.